### PR TITLE
OCM-4313: Update golang in ocm-sdk-go v1.21

### PR DESCRIPTION
**Pull request by machi1990**

### CODE REVIEW

**Comment by machi1990** [comment]
PTAL @zgalor @tzvatot @ciaranRoche once approved, I'd like to get https://github.com/openshift-online/ocm-sdk-go/pull/856 in as well and a new releases be created

**Comment by miguelsorianod** [comment]
The changes lgtm. Just a side question: Have you run `go mod tidy` after updating to Go 1.20?
If you did I am guessing there were some changes related to Go's management of dependencies, as I see several of them being added as indirect

**Comment by miguelsorianod** [comment]
> The changes lgtm. Just a side question: Have you run `go mod tidy` after updating to Go 1.20? If you did I am guessing there were some changes related to Go's management of dependencies between Go 1.16 and 1.20, as I see several of them being added as indirect

I am guessing it must have been in Go 1.17: https://tip.golang.org/doc/go1.17

**Comment by machi1990** [comment]
> The changes lgtm. Just a side question: Have you run `go mod tidy` after updating to Go 1.20? If you did I am guessing there were some changes related to Go's management of dependencies between Go 1.16 and 1.20, as I see several of them being added as indirect

Yes, I've run `go mod tidy` and almost all of the changes in the go.mod have been due to that.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift-online/ocm-sdk-go
 
-go 1.16
+go 1.20
```
Comment (zgalor) [inline go.mod:3]: why not 1.21?

File: .github/workflows/check-pull-request.yaml [hunk 2]
```diff
@@ -29,9 +29,8 @@ jobs:
     strategy:
       matrix:
         go:
-        - 1.18
-        - 1.19
-        - '1.20'
+        - "1.20"
+        - "1.21"
```
Comment (zgalor) [inline .github/workflows/check-pull-request.yaml:33]: why was this added?

Comment (machi1990) [reply .github/workflows/check-pull-request.yaml:33]: I added 1.20 & 1.21 here since I seem to remember that there are projects on 1.20 & 1.21 e.g  CS & AMS respecitvely, so I thought we could test against both

Comment (zgalor) [reply .github/workflows/check-pull-request.yaml:33]: and why not bump the sdk to 1.21?

Comment (tzvatot) [reply .github/workflows/check-pull-request.yaml:33]: I guess that will force any dependent project to upgrade too. Which is not too bad. It's mainly CS, AMS, OSL, AS, MSS, FM.

Comment (machi1990) [reply .github/workflows/check-pull-request.yaml:33]: Okay, let's give it a `go` to 1.21. 
At worst, dependent will be forced to update. Otherwise the go compiler will try to build the dependent package while ensuring the compatibleness promise.

Comment (machi1990) [reply .github/workflows/check-pull-request.yaml:32]: Updated

Comment (machi1990) [reply .github/workflows/check-pull-request.yaml:32]: > It's mainly CS, AMS, OSL, AS, MSS, FM.

@tzvatot @zgalor Just for information - there are other services using the ocm-sdk-go module e.g RHACS https://github.com/stackrox/acs-fleet-manager/blob/main/go.mod#L36

Comment (tzvatot) [reply .github/workflows/check-pull-request.yaml:32]: When dependent components would like to upgrade to latest OCM SDK, then they will need to update go latest Go as well.

I think that's reasonable. Can you please also run it with BU?

Comment (machi1990) [reply .github/workflows/check-pull-request.yaml:32]: I've tagged BU on the JIRA.

**Comment by machi1990** [comment]
@zgalor @tzvatot PTAL

**Review by tzvatot — APPROVED** [review]
